rfc/rfc_001_stamping-txs.md

```diff
@@ -0,0 +1,117 @@
+# RFC 048: Stamping transactions
+
+## Changelog
+
+* 11-09-2019: Initial draft
+
+## Context
+
+Here I am going to explain how we can decouple transactions from the block by stamping them.
+Stamping a transaction simply means putting the previous block hash in the header of a transaction.
+A *stamped transaction* unlike normal transaction has life time and it's only valid for a certain time (TTL).
+If `N` is the number of blocks after the stamp (or block hash), the *stamped transaction* after N blocks will be invalid.
+By stamping transactions we don't need to save the transactions inside a block and only keeping the transaction hash is fair enough.
+
+```
+ Stamped transaction
+
+ +----------------------------+
+ | STAMP                      |
+ +----------------------------+
+ | PAYLOAD                    |
+ |                            |
+ |                            |
+ +----------------------------+
+ | SIGNATURE                  |
+ |                            |
+ +----------------------------+
+```
+
+### How to validate a stamped transaction
+
+A stamped transaction easily can be verified off-chain.
+First we need to find the block with corresponding stamp (or block hash).
+If it doesn’t exist it means transaction is not valid (or probably not belong to this fork).
+If the block is found, we need to iterate over `N` block after to check if transaction hash is inside a block. Otherwise transaction is not valid.
```

Comment:
I suggest rewording to:

If the block is found at height S, we need to iterate over the txs in blocks `[S, S + N]` to find the transaction with this hash. Otherwise the transaction is invalid.

Reply:
Thanks. Please let me know if you have any suggestion on rewording the text.